### PR TITLE
Docs/storybookjsx

### PR DESCRIPTION
**Pull request by marielaures**
**Summary**

Add a new Storybook addon to display the example usage of every widget (see #60).
 
**Result**

Our users can now have a better idea of how the widgets are used. 
They can see the widget in action and get the same configuration as the one displayed in the story.
image

It also works for the 'playground' story where the props can be modified.
addonjsx

### CODE REVIEW

File: stories/Breadcrumb.stories.js [hunk 1]
```diff
@@ -1,51 +1,66 @@
...
-    </WrapWithHits>
-  ))
+    ),
+    { displayName: changeDisplayName, filterProps: filteredProps }
+  );
+
+stories
   .add('with custom component', () => (
     <WrapWithHits hasPlayground={true} linkedStoryGroup="Breadcrumb">
       <Breadcrumb
```
Comment (Haroenv) [inline stories/Breadcrumb.stories.js]: These last two don't have jsx

File: stories/SearchBox.stories.js [hunk 2]
```diff
@@ -1,65 +1,83 @@
...
+  .addWithJSX(
+    'with submit and reset components',
+    () => (
+      <WrapWithHits
+        searchBox={false}
+        hasPlayground={true}
+        linkedStoryGroup="SearchBox"
+      >
+        <SearchBox
+          submitComponent={<span>🔍</span>}
```
Comment (Haroenv) [inline stories/SearchBox.stories.js]: syntax highlighting was *off* in the live version here

**Comment by Haroenv** [comment]
The `jsx` tab is also visible on those that don't have it, a little confusing, because I happened to click on Conditional and Configure first 

- Conditional
- Configure
- Snippet

**Comment by marielaures** [comment]
Concerning the fact that we're displaying an empty JSX tab, I completely agree with you @Haroenv  but in some cases it can be even more confusing to output something that is either not relevant or messy (see example below):

image

It seems it is not possible to hide/display the panel conditionally. Happy to discuss it if you think of a better alternative!

File: package.json [hunk 3]
```diff
@@ -113,6 +113,7 @@
     "rheostat": "2.1.1",
     "sass-loader": "6.0.6",
     "stat-mode": "0.2.2",
+    "storybook-addon-jsx": "^5.0.0",
```
Comment (samouss) [inline package.json]: The dependency should be pin like the others.

**Comment by Haroenv** [comment]
Awesome 🎉 🌮

**Pull request merged**